src/components/Layout/Dropdown.astro

```diff
@@ -0,0 +1,163 @@
+---
+import IconCaret from '~icons/bi/caret-down-fill'
+interface Props {
+  inlineMenu?: boolean;
+  ariaLabel?: string;
+  class?: string;
+  trigger?: "click" | "hover";
+}
+
+// generate unique ids for each component so that we can connect the button to the dropdown without requiring consumer to provide an id
+// https://github.com/MicroWebStacks/astro-examples/blob/c278c3fc597bce013d627025a0c93758d72cc433/05_client-uid-counters/src/libs/utils.js#L8
+function uid() {
+  let date = (Date.now()).toString();
+  const sub = date.substring(date.length-6,date.length-1);
+  return sub+"_"+Math.floor(Math.random() * 10000)
+}
+
+let id = uid();
+let { inlineMenu, ariaLabel, class: className, trigger } = Astro.props; 
+---
+
+<div
+  data-trigger={trigger ?? "click"}
+  class:list={["dropdown", "js-dropdown", { "dropdown-inline js-dropdown-inline": inlineMenu }, className]}
+> 
+  <!-- 
+    Safari still has issues with focus on button click, see
+    https://bugs.webkit.org/show_bug.cgi?id=22261
+
+    As a result, this dropdown had issues closing because 
+    focusout events were not being fired
+    and that's why we have an "outside click" listener
+  -->
+  <button 
+    aria-controls={id} 
+    aria-label={ariaLabel} 
+    class="dropdown-toggle js-dropdown-toggle" 
+  >
+    <!-- 
+      The div and caret are in here because we want btn styles to apply
+      even with Astro's scoped styles 
+    -->
+    <div>
+      <slot name="toggle"/>
+    </div>
+    <IconCaret class="text-xs" />
+  </button>
+  <div 
+    id={id} 
+    class="dropdown-content z-nav-dropdown" 
+  >
+    <slot name="content"/>
+  </div>
+</div>
+
+<style>
+  @import '../../styles/global.css';
```

Comment:
@Aierie I don't think this import is needed since the base layout pulls it in, right? I tested it locally and seemed to work alright.

Reply:
When I run `pnpm dev` without the `@import` statement, the build completes and I can view the site. However, I get this error in the build logs:

```
 error   The `btn` class does not exist. If `btn` is a custom class, make sure it is defined within a `@layer` directive.
  File:
    /Users/michael/Personal/Code/tauri-docs/src/components/Layout/Dropdown.astro:66:5
  Code:
    64 |  
    65 |    .dropdown-toggle {
    66 |      @apply btn btn-inline text-left;
       |       ^
    67 |    }
    68 |
  Stacktrace:
CssSyntaxError: [postcss] /Users/michael/Personal/Code/tauri-docs/src/components/Layout/Dropdown.astro?astro&type=style&lang.css:11:5: The `btn` class does not exist. If `btn` is a custom class, make sure it is defined within a `@layer` directive.
```

Reply:
I think it's because of this: https://docs.astro.build/en/guides/integrations-guide/tailwind/#class-does-not-exist-with-apply-directives

But you also get a warning with the import on build due to the font being reimported I think:

```
./files/rubik-cyrillic-ext-400-normal.woff2 referenced in /Users/lorenzolewis/Developer/tauri-docs/src/components/Layout/Dropdown.astro?astro&type=style&index=0&lang.css didn't resolve at build time, it will remain unchanged to be resolved at runtime
```

If you remove the `.dropdown-toggle` styles and move `btn btn-inline text-left` to be inline directly in the `button` component then that should fix it.